dev/jobs/prep.sh

```diff
@@ -129,23 +142,25 @@ if [[ ${MAKE_PREPBUFR} = "YES" ]]; then
     # Do not fail on external errors
     set +eu
     "${HOMEobsproc}/jobs/JOBSPROC_GLOBAL_PREP" && true
-    err=$?
+    export err=$?
     if [[ ${err} -ne 0 ]]; then
-       echo "FATAL ERROR: Global prep job failed!"
-       exit 1
+       err_exit "JOBSPROC_GLOBAL_PREP job failed!"
     fi
-    set_strict
 
     # If creating NSSTBUFR was disabled, copy from DMPDIR if appropriate.
     if [[ ${MAKE_NSSTBUFR:-"NO"} = "NO" ]]; then
-        if [[ ${DONST} = "YES" ]]; then ${NCP} "${COMIN_OBSDMP}/${OPREFIX}nsstbufr" "${COMOUT_OBS}/${OPREFIX}nsstbufr"; fi
+        if [[ ${DONST} = "YES" ]]; then
+           cpreq "${COMINobsproc}/${OPREFIX}nsstbufr" "${COMOUT_OBS}/${OPREFIX}nsstbufr"
+        fi
     fi
 
 else
     if [[ ${ROTDIR_DUMP} = "NO" ]]; then
-        ${NCP} "${COMIN_OBSDMP}/${OPREFIX}prepbufr"               "${COMOUT_OBS}/${OPREFIX}prepbufr"
-        ${NCP} "${COMIN_OBSDMP}/${OPREFIX}prepbufr.acft_profiles" "${COMOUT_OBS}/${OPREFIX}prepbufr.acft_profiles"
-        if [[ ${DONST} = "YES" ]]; then ${NCP} "${COMIN_OBSDMP}/${OPREFIX}nsstbufr" "${COMOUT_OBS}/${OPREFIX}nsstbufr"; fi
+        cpreq "${COMINobsproc}/${OPREFIX}prepbufr" "${COMOUT_OBS}/${OPREFIX}prepbufr"
+        cpreq "${COMINobsproc}/${OPREFIX}prepbufr.acft_profiles" "${COMOUT_OBS}/${OPREFIX}prepbufr.acft_profiles"
+        if [[ ${DONST} = "YES" ]]; then
+           cpreq "${COMINobsproc}/${OPREFIX}nsstbufr" "${COMOUT_OBS}/${OPREFIX}nsstbufr"
```

Comment:
`cpfs` when copying to `COM`

```suggestion
        cpfs "${COMINobsproc}/${OPREFIX}prepbufr" "${COMOUT_OBS}/${OPREFIX}prepbufr"
        cpfs "${COMINobsproc}/${OPREFIX}prepbufr.acft_profiles" "${COMOUT_OBS}/${OPREFIX}prepbufr.acft_profiles"
        if [[ ${DONST} = "YES" ]]; then
           cpfs "${COMINobsproc}/${OPREFIX}nsstbufr" "${COMOUT_OBS}/${OPREFIX}nsstbufr"
```

Reply:
Lol pretty sure I had these as `cpfs` and then @aerorahul requested `cpreq`. But I agree, should be `cpfs`.

Reply:
`cpfs` also has a check that the copy was completed successfully, so it is an implied `cpreq` + `cp; mv` operation:

```
cp $1 $cpdstfile.cptmp

if [ $? -ne 0 ] ; then
   err_exit "$1 is missing or was not copied successfully."
fi

${FSYNC} $cpdstfile.cptmp
if [ $? -ne 0 ]; then
   >&2 echo "WARNING: ${FSYNC} $cpdstfile.cptmp failed."
fi

mv $cpdstfile.cptmp $cpdstfile
if [ $? -ne 0 ] ; then
   err_exit "$cpdstfile.cptmp is missing or was not moved successfully."
fi
```

Reply:
> However, when the GSI comes along in anal.sh, it will fail. So, by using cpreq here, we are catching the error much earlier.

I like that however, since we don't run the prep job in the GFS suite in ops (upstream obsproc does) we should have this handled within the analysis job if needed.

Reply:
Discussed offline, will use `cpfs`.